x-pack/platform/plugins/shared/triggers_actions_ui/public/common/alert_rule_from_vis_ui_action/alert_rule_from_vis_ui_action.tsx

```diff
@@ -64,175 +58,23 @@ export class AlertRuleFromVisAction implements Action<Context> {
-    const { state } = embeddable.serializeState().rawState.attributes ?? {};
-    const layers = (state?.datasourceStates?.textBased as TextBasedPersistedState | undefined)
-      ?.layers;
-    const [firstLayer] = Object.values(layers ?? {});
-    const { timeField = 'timestamp' } = firstLayer ?? { timeField: dataView?.timeFieldName };
-
-    // Generate additional query to append to the base ES|QL query from the visualization
-    const additionalQuery = buildAdditionalQuery(embeddableData);
-
-    // Assemble the full ES|QL code
-    let initialValues: Partial<RuleFormData<Partial<EsQueryRuleParams>>>;
-    if (query) {
-      const queryHeader = i18n.translate(
-        'xpack.triggersActionsUI.alertRuleFromVis.queryHeaderComment',
-        {
-          defaultMessage: 'Original ES|QL query derived from the visualization:',
-        }
-      );
-
-      initialValues = {
-        params: {
-          searchType: 'esqlQuery',
-          esqlQuery: {
-            esql: `// ${queryHeader}\n${query}\n${additionalQuery}`,
-          },
-          timeField,
-        },
-      };
-    } else {
-      const missingQueryComment = `// ${i18n.translate(
-        'xpack.triggersActionsUI.alertRuleFromVis.missingQueryComment',
-        {
-          defaultMessage: 'Unable to generate an ES|QL query from the visualization.',
-        }
-      )}`;
-      let esql = missingQueryComment;
-
-      if (dataView) {
-        const [index] = dataView.matchedIndices;
-        const esqlFromDataviewComment = `// ${i18n.translate(
-          'xpack.triggersActionsUI.alertRuleFromVis.esqlFromDataviewComment',
-          {
-            defaultMessage:
-              'Unable to automatically generate an ES|QL query that produces the same data as this visualization. You may be able to reproduce it manually using this data source:',
-          }
-        )}`;
-        const dataViewQuery = `FROM ${index}`;
-        esql = `${esqlFromDataviewComment}\n${dataViewQuery}\n${additionalQuery}`;
-      }
-      initialValues = {
-        params: {
-          searchType: 'esqlQuery',
-          esqlQuery: {
-            esql,
-          },
-          timeField,
-        },
-      };
-    }
-
-    const parentApi = embeddable.parentApi;
-
-    const overlayTracker = tracksOverlays(parentApi) ? parentApi : undefined;
-
-    const Component = await getRuleFlyoutComponent(
-      this.startDependencies,
-      this.ruleTypeRegistry,
-      this.actionTypeRegistry,
-      parentApi,
-      {
-        ...initialValues,
-        tags: [],
-        name: i18n.translate(
-          'xpack.triggersActionsUI.alertRuleFromVis.embeddable.alertRuleTitle.defaultName',
-          {
-            defaultMessage: 'Elasticsearch query rule from visualization',
-          }
-        ),
-      } as RuleFormData<EsQueryRuleParams>
-    );
-
-    embeddable.mountInlineFlyout(Component, overlayTracker, { dataTestSubj: 'lensAlertRule' });
+    await openLazyFlyout({
```

Comment:
openLazyFlyout is not async so no need to `await`

Reply:
All done! 
https://github.com/elastic/kibana/pull/226052/commits/99e8da9f4a8af6a10669d9d944c8bf1bb3c35fb3
https://github.com/elastic/kibana/pull/226052/commits/c89b3fa78a109bc0d8c2045099e4dd13f826c1e7